RFC-011-SWAP-Omni-Layer-basic.md

```diff
@@ -0,0 +1,308 @@
+<payload>
+```
+<!-- Refer to omni layer spec for simple send instead of this -->
+
+To then embed the Omni Layer Data received from omnicore in an `OP_RETURN` output, it MUST be prefixed it with the Omni Layer header: `6f6d6e69`.
+
+### Dust
+
+<!-- Dust actually refers to UTXOs that are not profitable to spend: https://eprint.iacr.org/2018/513 -->
+The reference implementation of the Bitcoin consensus, _bitcoind_ enforces that spendable UTXO MUST own enough value to be spent.
+<!-- Is this true? Link please.  -->
+If the value owned by a UTXO is less than the minimum spendable then it is what we commonly call dust and such transaction would be considered as *non-standard* and rejected.
+<!-- Yeah this I think is true. But if a non-standard transaction gets in I think it can be spent? -->
+
+For `bitcoind`, it is currently defined in term of `dustRelayFee` and it is set by default to `3000 satoshis-per-kilobytes`.
+The minimum value is 546 satoshis for a normal transaction and 294 satoshis for SegWit transactions.
+
+As [RFC-005](./RFC-005-SWAP-Bitcoin-basic.md) only defines SegWit HTLC then the former value is to be considered and will be referred as `min_sat` (294 Satoshis).
+<!-- ^I think this sentence isn't true if you use P2SH(P2WSH) but not sure -->
+
+`OP_RETURN` outputs are not spendable and hence SHOULD not be assigned any Bitcoin value.
+
+The only requirement is that spendable output MUST have at least `min_sat` Satoshis assigned.
+<!-- Is this really true? If you're a miner you can mine txs less than min_sat right? So it's a SHOULD I guess -->
+
+### Deployment transaction
+<!-- Put this under a section called Execution phase. See the other RFCs. -->
+<!-- Each section should start with how to execute the action and then how both parties can tell if the action has been executed -->
+
+The deployment transaction MUST be a simple send of the correct amount and kind of Omni Assets to the HTLC address.
+
+The HTLC address MUST either be:
+1. the P2WSH address of the HTLC, as described in [RFC-005](./RFC-005-SWAP-Bitcoin-basic.md)
+2. the P2SH of the P2WSH above <!-- Need to link to the BIP describing this -->
+
+(1) is preferred however, not all Omni Layer wallets implementation support bech32 addresses, hence, (2) can be used as a fallback.
+
+The simple send MUST NOT be done to the direct P2SH address of the HTLC.
+<!-- Maybe give a reason for this -->
+
+The party watching for the deployment transaction MUST consider both (1) and (2) addresses.
+
+Such party MUST check the validity of the transaction against Omni Layer Spec.
+Not all valid Bitcoin transactions containing Omni Data are valid Omni Layer transactions.
+
+For example, the following RPC call to omnicore can be used to verify the validity of a Omni Layer transaction:
+- `txid`: the transaction id of the transaction
+ ```
+$ [...] omni_gettransation <txid>
+{
+  [...]
+  "valid" : true
+  [...]
+}
+```
+
+<!-- we have to check the value that was transferred as well -->
```

Comment:
@LLFourn this is not necessary. if the transaction is a simple send to the correct recipient with the correct amount and is omni layer valid then we know that the asset has been transferred.

Reply:
Yes that's what I meant :)